ikarus/controlroutines/adaptivestepsizing.hh

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2021-2023 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#include <ikarus/solver/nonlinearsolver/solverinfos.hh>
+#include <ikarus/utils/pathfollowingfunctions.hh>
+
+namespace Ikarus {
+  template <typename ScalarType = double>
+  struct NoAdaptiveStepSizing {
+  public:
+    template <typename NonLinearOperator>
+    void operator()(const NonLinearSolverInformation&, Ikarus::SubsidiaryArgs&, const NonLinearOperator&) {}
+    [[nodiscard]] int targetIterations() const { return 0; }
+    void setTargetIterations(int) {}
+  };
+
+  template <typename ScalarType = double>
+  struct DefaultAdaptiveStepSizing {
+  public:
+    template <typename NonLinearOperator>
+    void operator()(const NonLinearSolverInformation& solverInfo, Ikarus::SubsidiaryArgs& subsidiaryArgs,
+                    const NonLinearOperator&) {
+      if (not((subsidiaryArgs.actualStep > 0) and (targetIterations_ != 0)))
+        DUNE_THROW(Dune::InvalidStateException,
+                   "For DefaultAdaptiveStepSizing, targetIterations_ should not be equal to 0 and actualStep should be "
```

Comment:
`targetIterations_` without underscore. The user does not know about member variables. Furthermore give a hint, such as "Did you forget about calling `setTargetIterations(..)` first?"